### PR TITLE
Enable misspell linter and fix spelling errors

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
I'd like to take the credit but it just a linter:
```
 golangci-lint run ./... --enable misspell --fix
```

**What this PR does / why we need it**:
Just some housekeeping

**Special notes for your reviewer**:
These could be automated by adding the [`misspell`](https://golangci-lint.run/usage/linters/#misspell) linter to the golangci-lint file.

**Release note**:

```release-note
Enable misspell linter and fix spelling errors in comments and strings
```

### CODE REVIEW

File: tests/utils/datavolume.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ const (
 	LargePhysicalDiskQcow = "http://cdi-file-host.%s/cirros-large-physical-size.qcow2"
 	// LargePhysicalDiskXz provides a test url for a cirros image with a large physical size, in RAW format, XZ-compressed
 	LargePhysicalDiskXz = "http://cdi-file-host.%s/cirros-large-physical-size.raw.xz"
-	// TarArchiveURL provides a test url for a tar achive file
+	// TarArchiveURL provides a test url for a tar achieve file
```
Comment (alromeros) [inline tests/utils/datavolume.go:74]: Typo: Archive

File: pkg/controller/datavolume/controller-base.go [hunk 2]
```diff
@@ -610,7 +610,7 @@ func (r *ReconcilerBase) handleStaticVolume(syncState *dvSyncState, log logr.Log
 	for _, v := range volumes {
 		if v == syncState.pvc.Spec.VolumeName {
 			pvcCpy := syncState.pvc.DeepCopy()
-			// handle as "populatedFor" going foreward
+			// handle as "populatedFor" going foreword
```
Comment (alromeros) [inline pkg/controller/datavolume/controller-base.go:613]: Typo: forward

File: tests/utils/datavolume.go (hunk 1 above)
Comment (EduardGomezEscandell) [reply tests/utils/datavolume.go]: Good catch

**Review by akalenyu — COMMENTED** [review]
Nice! should we enforce it going forward similarly to https://github.com/kubevirt/containerized-data-importer/pull/3097?

/approve

**Review by ShellyKa13 — APPROVED** [review]
/lgtm

**Comment by akalenyu** [comment]
Same failure as in https://github.com/kubevirt/containerized-data-importer/pull/2983#issuecomment-2036415808
/test pull-containerized-data-importer-e2e-hpp-previous

**Comment by akalenyu** [comment]
/hold
@EduardGomezEscandell feel free to cancel following https://github.com/kubevirt/containerized-data-importer/pull/3164#pullrequestreview-1984872120

**Comment by akalenyu** [comment]
> > should we enforce it going forward similarly to #3097?
> 
> I'd argue yes, do you prefer I do it here or on a follow-up PR?

It's fine both ways

**Comment by alromeros** [comment]
As a nitpick, I'd merge all the "fix typos" commits into a single one as we usually try splitting commits into logical units (in this case, one for the linter addition and another for the fixes would be great). We have a PR with commit guidelines but it's yet to be merged https://github.com/kubevirt/containerized-data-importer/pull/2752. I'd apply this to https://github.com/kubevirt/containerized-data-importer/pull/3171 too.

**Comment by EduardGomezEscandell** [comment]
Sounds good, I wanted to avoid force-pushing during the review but I can clean it up now.

**Comment by alromeros** [comment]
Thanks!
/lgtm

**Comment by EduardGomezEscandell** [comment]
Oops, apparently I introduced fresh new typos in my first PR :hand_over_mouth:

**Pull request merged**